config/default.yml

```diff
@@ -67,6 +67,7 @@ MissingRequiredTemplateFiles:
 
 UndefinedObject:
   enabled: true
+  ignore_snippets: true
```

Comment:
Should this be the opposite? default to false and can be turned on?

Reply:
`dont_ignore_snippets: false`? :p

Good point! renaming

Reply:
Actually, would it make more sense to name it `exclude_snippets: true` ?

To mimic the other one we already have: https://github.com/Shopify/theme-check/blob/master/config/default.yml#L33

Reply:
Actually I'm saying this should be opt-in in Dawn rather than by default to everyone.

Reply:
Oh I don't think this will roll. We want to use the same things internally that we'll promote externally.

So maybe this shouldn't even be an option? Always ignore the snippets.

Reply:
Right but what's the difference between a typo and an optional parameter in this case? Should you be flagged if you use a typo'ed argument?

Reply:
Yes I agree that we should. But I can't think of a way to allow optional parameters in snippets, and check for this at the same time.

I think we had two choices:
1. Force users to declare optional parameters (using `| default: ...`)
2. Stop checking snippets for this, because snippets are not

I wished we could stick w/ 1, but I agree it does make the code more complicated, as mentioned Slack.

And we're going w/ 2, because "we don't want to encourage the use of snippets as functions" anyways.

Maybe there's another solution, but I don't see it atm.

Reply:
Alright alright :p